core/framework.py

```diff
@@ -501,7 +501,9 @@ def _init_history(self, reborn=False, write=True):
 		# initialize history file
 		if not os.path.exists(history):
 			self._is_readable(history,'w').close()
-		if write:
+		if reborn:
```

Comment:
in _log_commands()  _init_history(True) makes reborn True which open file in write mode which deletes all previous entry in history during logging